From 48f4fb7d688e26b05c76e3d0fabdefc6c5f66241 Mon Sep 17 00:00:00 2001
From: yehui ren <renyehui@uniontech.com>
Date: Fri, 2 Jul 2021 03:42:26 +0000
Subject: [PATCH 1/4] cases/16.t: loongarch only supports eventfd2

Signed-off-by: yehui ren <renyehui@uniontech.com>
Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
---
 harness/cases/16.t | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/harness/cases/16.t b/harness/cases/16.t
index b36bbd2..4d24b04 100644
--- a/harness/cases/16.t
+++ b/harness/cases/16.t
@@ -18,8 +18,8 @@
 #define SYS_eventfd 318
 #elif defined(__alpha__)
 #define SYS_eventfd 478
-#elif defined(__aarch64__) || defined(__riscv)
-/* arm64 and riscv do not implement eventfd, only eventfd2 */
+#elif defined(__aarch64__) || defined(__loongarch__) || defined(__riscv)
+/* these recent architectures do not implement eventfd, only eventfd2 */
 #define USE_EVENTFD2
 #ifndef SYS_eventfd2
 #define SYS_eventfd2 19
-- 
2.33.0

