From 97076c8a55064e6f34483d41f6bffff3838f5b61 Mon Sep 17 00:00:00 2001
From: yehui ren <renyehui@uniontech.com>
Date: Fri, 2 Jul 2021 04:39:51 +0000
Subject: [PATCH 2/4] Add loongarch to supported architectures in libaio.spec

Signed-off-by: yehui ren <renyehui@uniontech.com>
Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
---
 libaio.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libaio.spec b/libaio.spec
index d1fa2cc..468be43 100644
--- a/libaio.spec
+++ b/libaio.spec
@@ -7,7 +7,7 @@ Group:  System Environment/Libraries
 Source: %{name}-%{version}.tar.gz
 BuildRoot: %{_tmppath}/%{name}-root
 # Fix ExclusiveArch as we implement this functionality on more architectures
-ExclusiveArch: i386 x86_64 ia64 s390 s390x ppc ppc64 ppc64pseries ppc64iseries alpha alphaev6 %{arm}
+ExclusiveArch: i386 x86_64 ia64 s390 s390x ppc ppc64 ppc64pseries ppc64iseries alpha alphaev6 %{arm} loongarch64
 
 %description
 The Linux-native asynchronous I/O facility ("async I/O", or "aio") has a
-- 
2.33.0

