From d892696468cb99d7d98b23b78bde942df0992d5a Mon Sep 17 00:00:00 2001
From: yehui ren <renyehui@uniontech.com>
Date: Fri, 2 Jul 2021 05:19:45 +0000
Subject: [PATCH 4/4] Use generic syscall number schema for loongarch

Signed-off-by: yehui ren <renyehui@uniontech.com>
Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
---
 src/syscall.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/syscall.h b/src/syscall.h
index b53da4c..d2a117b 100644
--- a/src/syscall.h
+++ b/src/syscall.h
@@ -27,7 +27,7 @@
 #include "syscall-arm.h"
 #elif defined(__sparc__)
 #include "syscall-sparc.h"
-#elif defined(__aarch64__) || defined(__riscv)
+#elif defined(__aarch64__) || defined(__loongarch__) || defined(__riscv)
 #include "syscall-generic.h"
 #else
 #warning "using system call numbers from sys/syscall.h"
-- 
2.33.0

