From 794fc24345e2868faf78cc3ee8dc8192e642f972 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Mon, 9 Aug 2021 18:48:52 +0800
Subject: [PATCH] Add symbol version for loongarch64

---
 lib/libcrypt.minver | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/libcrypt.minver b/lib/libcrypt.minver
index 445b946..952f0ef 100644
--- a/lib/libcrypt.minver
+++ b/lib/libcrypt.minver
@@ -59,6 +59,7 @@ GLIBC_2.3    kfreebsd.*gnu   i[3-9]86
 ERROR        kfreebsd.*gnu   .
 
 # Linux with GNU libc
+GLIBC_2.33   linux.*gnu      loongarch64
 GLIBC_2.33   linux.*gnu      riscv32
 GLIBC_2.32   linux.*gnu      arc
 GLIBC_2.29   linux.*gnu      csky
-- 
2.30.1

