From 28c6d65b0752ec15fc0c66225bd7f9fc12c6f2ba Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Sat, 11 Sep 2021 12:54:28 +0800
Subject: [PATCH] Add support for LoongArch

Signed-off-by: WANG Xuerui <git@xen0n.name>
---
 src/local-elf.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/local-elf.h b/src/local-elf.h
index 98a2962..1dd3b42 100644
--- a/src/local-elf.h
+++ b/src/local-elf.h
@@ -124,6 +124,16 @@
 #define ELF_TARG_CLASS	ELFCLASS64
 #define ELF_TARG_DATA	ELFDATA2LSB
 
+#elif defined(__loongarch__)
+
+#define ELF_TARG_MACH	EM_LOONGARCH
+#if defined(__loongarch64)
+#define ELF_TARG_CLASS	ELFCLASS64
+#else
+#error Unsupported ELF class
+#endif
+#define ELF_TARG_DATA	ELFDATA2LSB
+
 #elif defined(__m32r__)
 
 #define ELF_TARG_MACH	EM_M32R
-- 
2.33.0

