From bfdfe556c0e3e310d9d830a6eb7c19327f38cddc Mon Sep 17 00:00:00 2001
From: caiyinyu <caiyinyu@loongson.cn>
Date: Tue, 28 Dec 2021 10:55:24 +0800
Subject: [PATCH 01/32] LoongArch: Update NEWS and README for the LoongArch
 port.

---
 NEWS   | 2 ++
 README | 1 +
 2 files changed, 3 insertions(+)

diff --git a/NEWS b/NEWS
index faa7ec1871..e5f8db1d3f 100644
--- a/NEWS
+++ b/NEWS
@@ -9,6 +9,8 @@ Version 2.35
 
 Major new features:
 
+* Add LoongArch Port.  Loongson Ltd.
+
 * Unicode 14.0.0 Support: Character encoding, character type info, and
   transliteration tables are all updated to Unicode 14.0.0, using
   generator scripts contributed by Mike FABIAN (Red Hat).
diff --git a/README b/README
index 3d5a78ccff..1c03949dfe 100644
--- a/README
+++ b/README
@@ -31,6 +31,7 @@ The GNU C Library supports these configurations for using Linux kernels:
 	i[4567]86-*-linux-gnu
 	x86_64-*-linux-gnu	Can build either x86_64 or x32
 	ia64-*-linux-gnu
+	loongarch64-*-linux-gnu
 	m68k-*-linux-gnu
 	microblaze*-*-linux-gnu
 	mips-*-linux-gnu
-- 
2.35.1

