From c24fa1b526eef4993e5a6434f4f657c789b4e6fd Mon Sep 17 00:00:00 2001
From: Fuxin Zhang <fxzhang@ict.ac.cn>
Date: Sat, 10 Dec 2022 12:45:57 +0000
Subject: [PATCH] fix compilation on loongarch: __int128__ to __int128

Signed-off-by: Fuxin Zhang <fxzhang@ict.ac.cn>
---
 src/mpfr-longlong.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mpfr-longlong.h b/src/mpfr-longlong.h
index 8e71eea79..43c665b06 100644
--- a/src/mpfr-longlong.h
+++ b/src/mpfr-longlong.h
@@ -1168,7 +1168,7 @@ extern UWtype __MPN(udiv_qrnnd) (UWtype *, UWtype, UWtype, UWtype);
   do {									\
     UDItype __u = (u), __v = (v);					\
     (w0) = __u * __v;							\
-    (w1) = (unsigned __int128__) __u * __v >> 64;			\
+    (w1) = (unsigned __int128) __u * __v >> 64;			\
   } while (0)
 #endif
 
-- 
2.39.0

