From 289f73c5d24a55399f4a2fb5be648840261c995a Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Sun, 3 Oct 2021 14:55:27 +0800
Subject: [PATCH 2/3] gen_funcs.sh: support probing if glibc is 2.34 or later

Signed-off-by: WANG Xuerui <git@xen0n.name>
---
 gen_funcs.sh | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/gen_funcs.sh b/gen_funcs.sh
index 500caf2..e110519 100755
--- a/gen_funcs.sh
+++ b/gen_funcs.sh
@@ -239,6 +239,34 @@ is_glibc() {
 	echo "${is_glibc}"
 }
 
+is_glibc_2_34_or_later() {
+	if ! isTrue "$(is_glibc)"; then
+		echo "no"
+		return
+	fi
+
+	local ver="$(getconf GNU_LIBC_VERSION)"
+	ver="${ver##glibc }"
+
+	local ver_major="${ver%%.*}"
+	ver="${ver##${ver_major}.}"
+
+	local ver_minor="${ver%%.*}"
+
+	# Don't bother checking 1.x at all.
+	if [[ $ver_major -gt 2 ]]; then
+		echo "yes"
+		return
+	fi
+
+	if [[ $ver_minor -ge 34 ]]; then
+		echo "yes"
+		return
+	fi
+
+	echo "no"
+}
+
 is_gzipped() {
 	[[ ${#} -ne 1 ]] \
 		&& gen_die "$(get_useful_function_stack "${FUNCNAME}")Invalid usage of ${FUNCNAME}(): Function takes exactly one argument (${#} given)!"
-- 
2.34.0

