From 6916a427ae0f19452d017ca42318543a03cf3e88 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Wed, 1 Sep 2021 14:33:40 +0800
Subject: [PATCH 8/8] udev: add loongarch64 to list of DMI-capable
 architectures

---
 src/udev/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/udev/meson.build b/src/udev/meson.build
index 4e80f9bfd7..f2eb0b2c60 100644
--- a/src/udev/meson.build
+++ b/src/udev/meson.build
@@ -129,7 +129,7 @@ udev_id_progs = [['ata_id/ata_id.c'],
                   'mtd_probe/mtd_probe.h',
                   'mtd_probe/probe_smartmedia.c']]
 
-dmi_arches = ['x86', 'x86_64', 'aarch64', 'arm', 'ia64', 'mips']
+dmi_arches = ['x86', 'x86_64', 'aarch64', 'arm', 'ia64', 'loongarch64', 'mips']
 if dmi_arches.contains(host_machine.cpu_family())
         udev_id_progs += [['dmi_memory_id/dmi_memory_id.c']]
 endif
-- 
2.35.1

