From 464d51a8d841cf67c9a3f24fb51362f4ba92b0b6 Mon Sep 17 00:00:00 2001
From: Chenghua Xu <xuchenghua@loongson.cn>
Date: Sun, 15 Aug 2021 10:40:59 -0400
Subject: [PATCH 20/55] Adjust EM_NUM to 259.

---
 elf/elf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elf/elf.h b/elf/elf.h
index 5b5a509c6d..c18207b52d 100644
--- a/elf/elf.h
+++ b/elf/elf.h
@@ -360,7 +360,7 @@ typedef struct
 #define EM_CSKY		252     /* C-SKY */
 #define	EM_LOONGARCH	258	/* LoongArch */
 
-#define EM_NUM		253
+#define EM_NUM		259
 
 /* Old spellings/synonyms.  */
 
-- 
2.34.0

