From d98f84ece7d9da2c2c4e9fe9b6df29ecf94e62c6 Mon Sep 17 00:00:00 2001
From: caiyinyu <caiyinyu@loongson.cn>
Date: Sun, 28 Nov 2021 17:58:11 +0800
Subject: [PATCH 40/51] update README

---
 README | 1 +
 1 file changed, 1 insertion(+)

diff --git a/README b/README
index d0f0edb393..f0812d7a80 100644
--- a/README
+++ b/README
@@ -31,6 +31,7 @@ The GNU C Library supports these configurations for using Linux kernels:
 	i[4567]86-*-linux-gnu
 	x86_64-*-linux-gnu	Can build either x86_64 or x32
 	ia64-*-linux-gnu
+	loongarch64-*-linux-gnu
 	m68k-*-linux-gnu
 	microblaze*-*-linux-gnu
 	mips-*-linux-gnu
-- 
2.34.0

