From 91002d47072ceb961756cc6704d19cdaf04331ef Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Wed, 22 Dec 2021 01:10:05 +0800
Subject: [PATCH 55/55] fix TCB_ALIGNMENT not defined on LoongArch

---
 sysdeps/loongarch/nptl/pthreaddef.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sysdeps/loongarch/nptl/pthreaddef.h b/sysdeps/loongarch/nptl/pthreaddef.h
index f13c947917..bf3b2471e0 100644
--- a/sysdeps/loongarch/nptl/pthreaddef.h
+++ b/sysdeps/loongarch/nptl/pthreaddef.h
@@ -28,5 +28,8 @@
 /* Minimal stack size after allocating thread descriptor and guard size.  */
 #define MINIMAL_REST_STACK 2048
 
+/* Alignment requirement for TCB.  */
+#define TCB_ALIGNMENT 16
+
 /* Location of current stack frame.  */
 #define CURRENT_STACK_FRAME __builtin_frame_address (0)
-- 
2.34.0

