From dc7f1f8d2b6ccd919dbb19601600064ed590bc83 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Tue, 28 Sep 2021 14:41:55 +0800
Subject: [PATCH 3/8] lower RELOC_STACK_MAX to 16

---
 util/grub-mkimagexx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/grub-mkimagexx.c b/util/grub-mkimagexx.c
index 5dfb01c06..013a7d420 100644
--- a/util/grub-mkimagexx.c
+++ b/util/grub-mkimagexx.c
@@ -61,7 +61,7 @@
 #endif
 #endif
 
-#define RELOC_STACK_MAX 1024
+#define RELOC_STACK_MAX 16
 
 /* These structures are defined according to the CHRP binding to IEEE1275,
    "Client Program Format" section.  */
-- 
2.33.0

