From 91edffd89114bbe35242fa3f2e71283b03672958 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Tue, 28 Sep 2021 21:29:56 +0800
Subject: [PATCH 6/8] fix unrecognized -march option

---
 conf/Makefile.common | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/conf/Makefile.common b/conf/Makefile.common
index bcbf95156..fec1fd7c6 100644
--- a/conf/Makefile.common
+++ b/conf/Makefile.common
@@ -21,8 +21,8 @@ if COND_powerpc_ieee1275
   CFLAGS_PLATFORM += -mcpu=powerpc
 endif
 if COND_loongarch64
-  CFLAGS_PLATFORM += -fno-strict-aliasing -march=loongarch -mabi=lp64 -fno-plt -Wa,-mla-global-with-pcrel
-  CPPFLAGS_PLATFORM = -fno-strict-aliasing -march=loongarch -mabi=lp64 -fno-plt -Wa,-mla-global-with-pcrel
+  CFLAGS_PLATFORM += -fno-strict-aliasing -mabi=lp64s -fno-plt -Wa,-mla-global-with-pcrel
+  CPPFLAGS_PLATFORM = -fno-strict-aliasing -mabi=lp64s -fno-plt -Wa,-mla-global-with-pcrel
 endif
 
 # Other options
-- 
2.33.0

