From 30459fe4efcf3113509407e94edd452f02fe39a8 Mon Sep 17 00:00:00 2001
From: GDB Administrator <gdbadmin@sourceware.org>
Date: Mon, 25 Oct 2021 00:00:17 +0000
Subject: [PATCH 09/97] Automatic date update in version.in

---
 bfd/version.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bfd/version.h b/bfd/version.h
index b8fefffa200..4747b0cb6c0 100644
--- a/bfd/version.h
+++ b/bfd/version.h
@@ -16,7 +16,7 @@
 
    In releases, the date is not included in either version strings or
    sonames.  */
-#define BFD_VERSION_DATE 20211024
+#define BFD_VERSION_DATE 20211025
 #define BFD_VERSION @bfd_version@
 #define BFD_VERSION_STRING  @bfd_version_package@ @bfd_version_string@
 #define REPORT_BUGS_TO @report_bugs_to@
-- 
2.34.0

