From d98c63919a079429b3f218c81ba03b04fd78df02 Mon Sep 17 00:00:00 2001
From: GDB Administrator <gdbadmin@sourceware.org>
Date: Tue, 9 Nov 2021 00:00:18 +0000
Subject: [PATCH 75/97] Automatic date update in version.in

---
 bfd/version.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bfd/version.h b/bfd/version.h
index 6b640021c55..56f5ba1668d 100644
--- a/bfd/version.h
+++ b/bfd/version.h
@@ -16,7 +16,7 @@
 
    In releases, the date is not included in either version strings or
    sonames.  */
-#define BFD_VERSION_DATE 20211108
+#define BFD_VERSION_DATE 20211109
 #define BFD_VERSION @bfd_version@
 #define BFD_VERSION_STRING  @bfd_version_package@ @bfd_version_string@
 #define REPORT_BUGS_TO @report_bugs_to@
-- 
2.34.0

