From bceeb329652840890bfea21c35804d24c81e4134 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <xen0n@gentoo.org>
Date: Tue, 3 Sep 2024 15:39:18 +0800
Subject: [PATCH 5/8] enable webrtc for loongarch64 in toolkit/moz.configure

Signed-off-by: WANG Xuerui <xen0n@gentoo.org>
---
 toolkit/moz.configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index bd6648a89e71c..109aa81d04227 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -1456,6 +1456,7 @@ def webrtc_default(target):
         "aarch64",
         "x86",
         "ia64",
+        "loongarch64",
         "mips32",
         "mips64",
         "ppc",
-- 
2.48.1

